src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java

```diff
@@ -311,11 +312,25 @@ ResultHandle getOutgoingResultHandle() {
     @Override
     public ResultHandle load(String val) {
         Objects.requireNonNull(val);
-        if (val.length() > 65535) {
-            //TODO: we could auto split this, but I don't think we really want strings this long
-            throw new IllegalArgumentException("Cannot load strings larger than " + 65535 + " bytes");
+        int length = val.length();
+        if (length > MAX_STRING_LENGTH) {
+            int div = length / MAX_STRING_LENGTH;
+            int mod = length % MAX_STRING_LENGTH;
+
+            Gizmo.StringBuilderGenerator sb = Gizmo.newStringBuilder(this, length);
+            for (int i = 0; i < div; i++) {
+                int beginIndex = i * MAX_STRING_LENGTH;
+                int endIndex = beginIndex + MAX_STRING_LENGTH;
+                sb.append(val.substring(beginIndex, endIndex));
+            }
+            if (mod > 0) {
+                sb.append(val.substring(div * MAX_STRING_LENGTH));
+            }
+            return sb.callToString();
+
+        } else {
+            return new ResultHandle("Ljava/lang/String;", this, val);
```

Comment:
I think that it would be worth mentioning this implementation details in the javadoc of this method, i.e. that a string < 65535 is stored in the constant pool while a string > 65535 is stored in the heap. A new string is constructed using the `StringBuilder` in the context of the current method body and the reference is returned.

Reply:
@mkouba : https://github.com/quarkusio/gizmo/pull/152